build/apple_release/scripts/download-signed-mac-OS-binaries.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+cliExecutableName=$1
+releaseVersion=$2
+goarch=$3
+GITHUB_ACCESS_TOKEN=$4
+
+# This script downloads signed macOS binaries for a specific version and architecture.
+
+# Function to retrieve the specific artifact URL with retries
+get_specific_artifact_url_with_retries() {
+    local max_retries=4
+    local cooldown=15 # Cooldown in seconds between retries
+    local retry_count=0
+
+    while [ $retry_count -lt $max_retries ]; do
+        # Fetch the list of artifacts from GitHub
+        response=$(curl -L \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer $GITHUB_ACCESS_TOKEN" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
```

Comment:
According to the [documentation](https://docs.github.com/en/rest/about-the-rest-api/api-versions?apiVersion=2022-11-28#about-api-versioning):
> When a new REST API version is released, the previous API version will be supported for at least 24 more months following the release of the new API version.

Let's use the latest API